engines/tot/engine.cpp

```diff
@@ -0,0 +1,5512 @@
+							} break;
+							}
+							_currentRoomData->screenLayers[0].bitmapSize = 892;
+							_currentRoomData->screenLayers[0].coordx = 217;
+							_currentRoomData->screenLayers[0].coordy = 48;
+							_currentRoomData->screenLayers[0].depth = 1;
+							_graphics->restoreBackground();
+							assembleScreen();
+							_graphics->drawScreen(_sceneBackground);
+							animateOpen2(0, 1);
+							updateInventory(usedObjectIndex);
+							drawInventory();
+							_mouse->show();
+						} else {
+
+							_niche[1][_niche[1][3]] = _curObject.code;
+							_niche[0][3] += 1;
+							_niche[1][3] -= 1;
+							drawText(_curObject.useTextRef);
+							_mouse->hide();
+							animateGive(0, 1);
+
+							switch (_curObject.code) {
+							case 561:
+								readBitmap(1381982, _screenLayers[_curObject.depth - 1],
+										   892, 319);
+								break;
+							case 615:
+								readBitmap(1381090, _screenLayers[_curObject.depth - 1],
+										   892, 319);
+								break;
+							}
+							_graphics->restoreBackground();
+							assembleScreen();
+							_graphics->drawScreen(_sceneBackground);
+							animateOpen2(0, 1);
+							updateInventory(usedObjectIndex);
+							drawInventory();
+							_currentRoomData->screenObjectIndex[8]->objectName = "                    ";
+							_sound->playVoc("PLATAF", 375907, 14724);
+							switch (_niche[1][_niche[1][3]]) {
+							case 0: {
+								_currentRoomData->screenObjectIndex[8]->objectName = getObjectName(4);
+								nicheAnimation(1, 1399610);
+								_currentRoomData->screenLayers[0].bitmapPointer = 1399610;
+							} break;
+							case 561: {
+								_currentRoomData->screenObjectIndex[8]->objectName = getObjectName(5);
+								nicheAnimation(1, 1381982);
+								_currentRoomData->screenLayers[0].bitmapPointer = 1381982;
+							} break;
+							case 615: {
+								_currentRoomData->screenObjectIndex[8]->objectName = getObjectName(7);
+								nicheAnimation(1, 1381090);
+								_currentRoomData->screenLayers[0].bitmapPointer = 1381090;
+							} break;
+							case 622: {
+								_currentRoomData->screenObjectIndex[8]->objectName = getObjectName(8);
+								nicheAnimation(1, 1400502);
+								_currentRoomData->screenLayers[0].bitmapPointer = 1400502;
+							} break;
+							case 623: {
+								_currentRoomData->screenObjectIndex[8]->objectName = getObjectName(9);
+								nicheAnimation(1, 1398718);
+								_currentRoomData->screenLayers[0].bitmapPointer = 1398718;
+							} break;
+							}
+							_currentRoomData->screenLayers[0].bitmapSize = 892;
+							_currentRoomData->screenLayers[0].coordx = 217;
+							_currentRoomData->screenLayers[0].coordy = 48;
+							_currentRoomData->screenLayers[0].depth = 1;
+							_mouse->show();
+							updateAltScreen(20);
+						}
+					} else {
+						drawText(Random(11) + 1022);
+					}
+				} break;
+				}
+			} break;
+			case 583: {
+
+				long offset = getOffsetsByCurrentLanguage()[20];
```

Comment:
```suggestion
				int32 offset = getOffsetsByCurrentLanguage()[20];
```
and in other similar places